### PR TITLE
Fill palette

### PR DESCRIPTION
**Pull request by prad1998**

### CODE REVIEW

File: JHotDraw/pom.xml [hunk 1]
```diff
@@ -51,6 +51,29 @@
...
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>4.13.1</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-core</artifactId>
+            <version>3.6.0</version>
+        </dependency>
```
Comment (karlh18) [inline JHotDraw/pom.xml:65]: Delete duplicate dependencies

File: JHotDraw/pom.xml [hunk 2]
```diff
@@ -51,6 +51,29 @@
...
+            <artifactId>jgiven-junit</artifactId>
+            <version>1.0.0-RC6</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.assertj</groupId>
+            <artifactId>assertj-core</artifactId>
+            <version>3.18.1</version>
+            <scope>test</scope>
         </dependency>
```
Comment (karlh18) [inline JHotDraw/pom.xml:77]: Delete Duplicated Dependencies

Comment (karlh18) [reply JHotDraw/pom.xml:77]: Remove duplicated dependencies such that we only use the latest version

File: JHotDraw/src/main/java/org/jhotdraw/draw/action/ButtonFactory.java [hunk 3]
```diff
@@ -573,6 +573,8 @@ public static JPopupButton createEditorColorButton(
         popupButton.setAction(
                 new DefaultAttributeAction(editor, attributeKey, defaultAttributes),
                 new Rectangle(0, 0, 22, 22));
+       // ############################################################################# 
+
```
Comment (karlh18) [inline JHotDraw/src/main/java/org/jhotdraw/draw/action/ButtonFactory.java:577]: delete?

Comment (karlh18) [reply JHotDraw/src/main/java/org/jhotdraw/draw/action/ButtonFactory.java:577]: The //#### comment that is

File: JHotDraw/src/main/java/org/jhotdraw/draw/action/ButtonFactory.java [hunk 4]
```diff
@@ -601,6 +603,7 @@ public static JPopupButton createEditorColorButton(
                     editor,
                     attributes,
                     labels.getToolTipTextProperty("attribute.color.noColor"),
+        // #############################################################################
```
Comment (karlh18) [inline JHotDraw/src/main/java/org/jhotdraw/draw/action/ButtonFactory.java:606]: Delete comment //#########

File: JHotDraw/src/main/java/org/jhotdraw/gui/plaf/palette/PaletteButtonBorder.java [hunk 5]
```diff
@@ -63,45 +69,21 @@ public void paintBorder(AbstractButton c, Graphics gr, int x, int y, int width,
...
-            insets = new Insets(3, 3, 3, 2);
-        } else {
-            insets = new Insets(3, 3, 3, 3);
-        }
-        return insets;
-    }
 
 
-    public boolean isBorderOpaque() {
-        return true;
```
Comment (karlh18) [inline JHotDraw/src/main/java/org/jhotdraw/gui/plaf/palette/PaletteButtonBorder.java:105]: Why were these deleted?

File: JHotDraw/src/main/java/org/jhotdraw/text/ColorFormatter.java [hunk 6]
```diff
@@ -264,11 +264,16 @@ public String valueToString(Object value) throws ParseException {
 
             switch (outputFormat) {
                 case RGB_HEX:
-                    str = "000000" + Integer.toHexString(c.getRGB() & 0xffffff);
+                    str = "000000" + Integer.toHexString(c.getRGB() & 0xffffff);
```
Comment (karlh18) [inline JHotDraw/src/main/java/org/jhotdraw/text/ColorFormatter.java:267]: Possible to delete the souts ?

File: JHotDraw/src/main/java/org/jhotdraw/samples/svg/gui/FillToolBar.java [hunk 7]
```diff
@@ -35,15 +36,21 @@
...
- * creation tool is active. 
- * <br>1.1 2008-03-26 Don't draw button borders. 
+ * creation tool is active.
+ * <br>1.1 2008-03-26 Don't draw button borders.
  * <br>1.0 May 1, 2007 Created.
  */
 public class FillToolBar extends AbstractToolBar {
 
     private SelectionComponentDisplayer displayer;
+    JPanel p;
```
Comment (karlh18) [inline JHotDraw/src/main/java/org/jhotdraw/samples/svg/gui/FillToolBar.java:46]: Can't you just move the JPanel p, down to be declared like the other JPanels found just beneath it?

File: JHotDraw/src/main/java/org/jhotdraw/samples/svg/gui/FillToolBar.java [hunk 8]
```diff
@@ -66,175 +73,217 @@ public void setEditor(DrawingEditor newValue) {
...
-                    gbc.anchor = GridBagConstraints.FIRST_LINE_START;
-                    p.add(p2, gbc);
-                    gbc = new GridBagConstraints();
-                    gbc.gridy = 2;
-                    gbc.weighty = 1f;
-                    gbc.anchor = GridBagConstraints.FIRST_LINE_START;
-                    p.add(p3, gbc);
-                }
-                break;
+        p = new JPanel(); p1 = new JPanel(new GridBagLayout()); p2 = new JPanel(new GridBagLayout()); p3 = new JPanel(new GridBagLayout());
```
Comment (karlh18) [inline JHotDraw/src/main/java/org/jhotdraw/samples/svg/gui/FillToolBar.java:76]: possible to separate the initializations into each row to increase readability? One option is possibly to add it into the constructor since you already initialized  the Resource Labels there?

File: JHotDraw/src/main/java/org/jhotdraw/samples/svg/gui/FillToolBar.java [hunk 9]
```diff
@@ -66,175 +73,217 @@ public void setEditor(DrawingEditor newValue) {
...
+        labels = ResourceBundleUtil.getBundle("org.jhotdraw.samples.svg.Labels");
```
Comment (karlh18) [inline JHotDraw/src/main/java/org/jhotdraw/samples/svg/gui/FillToolBar.java:77]: I think you can delete this line because you already initialize the labels in the constructor

File: JHotDraw/src/main/java/org/jhotdraw/samples/svg/gui/FillToolBar.java [hunk 10]
```diff
@@ -66,175 +73,217 @@ public void setEditor(DrawingEditor newValue) {
...
+        if (state == 1) {
+            panelset();
+        } else if (state == 2) {
+           gbc = new GridBagConstraints();
+            setp();
```
Comment (karlh18) [inline JHotDraw/src/main/java/org/jhotdraw/samples/svg/gui/FillToolBar.java:82]: Can you rename this method ? Not clear enough what it does -  setPanel or  SetPanelLayout  perhaps?

File: JHotDraw/src/main/java/org/jhotdraw/samples/svg/gui/FillToolBar.java [hunk 11]
```diff
@@ -66,175 +73,217 @@ public void setEditor(DrawingEditor newValue) {
...
+    public void colorFieldButton() {
+
+        Map<AttributeKey, Object> defaultAttributes = new HashMap<AttributeKey, Object>();
+        FILL_GRADIENT.set(defaultAttributes, null);
+
+        AbstractButton btn = ButtonFactory.createSelectionColorButton(editor,
+                FILL_COLOR, ButtonFactory.HSV_COLORS, ButtonFactory.HSV_COLORS_COLUMN_COUNT,
+                "attribute.fillColor", labels, defaultAttributes, new Rectangle(3, 3, 10, 10));
+
+        // Metode med colorfield
```
Comment (karlh18) [inline JHotDraw/src/main/java/org/jhotdraw/samples/svg/gui/FillToolBar.java:236]: remove comment or make it english

File: JHotDraw/src/main/java/org/jhotdraw/samples/svg/gui/FillToolBar.java [hunk 12]
```diff
@@ -66,175 +73,217 @@ public void setEditor(DrawingEditor newValue) {
...
+        gbc.anchor = GridBagConstraints.FIRST_LINE_START;
+        p.add(p2, gbc);
+        gbc = new GridBagConstraints();
+        gbc.gridy = 2;
+        gbc.weighty = 1f;
+        gbc.anchor = GridBagConstraints.FIRST_LINE_START;
+        p.add(p3, gbc);
+    }
+
+    public void opacityhandler(JAttributeTextField<Double> opacityField) {
```
Comment (karlh18) [inline JHotDraw/src/main/java/org/jhotdraw/samples/svg/gui/FillToolBar.java:199]: By reading this method it seems you actually create a button, which you thereafter add a handler ? Perhaps should be reanmed createOpacityButton or similar

File: JHotDraw/src/main/java/org/jhotdraw/samples/svg/gui/FillToolBar.java [hunk 13]
```diff
@@ -66,175 +73,217 @@ public void setEditor(DrawingEditor newValue) {
...
+            // Fill color field and button     
+            colorFieldButton();
+            opacity();
+            // Add horizontal strips   
+            gbcstrips();
         }
         return p;
     }
 
+    public void fillcolor(AbstractButton btn) {
```
Comment (karlh18) [inline JHotDraw/src/main/java/org/jhotdraw/samples/svg/gui/FillToolBar.java:92]: In general the methods could be renamed something more meaningful because you're actually creating a button here

Comment (karlh18) [reply JHotDraw/src/main/java/org/jhotdraw/samples/svg/gui/FillToolBar.java:92]: In this case you're creating a  "SelectionColorButton" which can be found on line 95

**Review by karlh18 — CHANGES_REQUESTED** [review]
In general fine, there are some things to delete, and change such as the name of the extracted methods found in FillToolBar class. Instead of calling a method that creates a button fillColor() a more suitable name might be createColorSelectionButton or similar

File: JHotDraw/src/main/java/org/jhotdraw/gui/plaf/palette/PaletteButtonBorder.java (hunk 5 above)
Comment (prad1998) [reply JHotDraw/src/main/java/org/jhotdraw/gui/plaf/palette/PaletteButtonBorder.java:105]: A method of Border interface that is used in a superclass instead of the subclasses

**Review by karlh18 — APPROVED** [review]
The requested changes are fixed and approved
